exercises/practice/kindergarten-garden/test_kindergarten_garden.c

```diff
@@ -0,0 +1,164 @@
+{
+   const plant_t expected[] = { VIOLETS, RADISHES, VIOLETS, RADISHES };
+   test_kindergarten_garden(
+       "VRCGVVRVCGGCCGVRGCVCGCGV\nVRCCCGCRRGVCGCRVVCVGCGCV", "Alice", expected);
+}
+
+static void test_full_garden_for_bob_second_student_s_garden(void)
+{
+   const plant_t expected[] = { CLOVER, GRASS, CLOVER, CLOVER };
+   test_kindergarten_garden(
+       "VRCGVVRVCGGCCGVRGCVCGCGV\nVRCCCGCRRGVCGCRVVCVGCGCV", "Bob", expected);
+}
+
+static void test_full_garden_for_charlie(void)
+{
+   const plant_t expected[] = { VIOLETS, VIOLETS, CLOVER, GRASS };
+   test_kindergarten_garden(
+       "VRCGVVRVCGGCCGVRGCVCGCGV\nVRCCCGCRRGVCGCRVVCVGCGCV", "Charlie",
+       expected);
+}
+
+static void test_full_garden_for_david(void)
+{
+   const plant_t expected[] = { RADISHES, VIOLETS, CLOVER, RADISHES };
+   test_kindergarten_garden(
+       "VRCGVVRVCGGCCGVRGCVCGCGV\nVRCCCGCRRGVCGCRVVCVGCGCV", "David", expected);
+}
+
+static void test_full_garden_for_eve(void)
+{
+   const plant_t expected[] = { CLOVER, GRASS, RADISHES, GRASS };
+   test_kindergarten_garden(
+       "VRCGVVRVCGGCCGVRGCVCGCGV\nVRCCCGCRRGVCGCRVVCVGCGCV", "Eve", expected);
+}
+
+static void test_full_garden_for_fred(void)
+{
+   const plant_t expected[] = { GRASS, CLOVER, VIOLETS, CLOVER };
+   test_kindergarten_garden(
+       "VRCGVVRVCGGCCGVRGCVCGCGV\nVRCCCGCRRGVCGCRVVCVGCGCV", "Fred", expected);
+}
+
+static void test_full_garden_for_ginny(void)
+{
+   const plant_t expected[] = { CLOVER, GRASS, GRASS, CLOVER };
+   test_kindergarten_garden(
+       "VRCGVVRVCGGCCGVRGCVCGCGV\nVRCCCGCRRGVCGCRVVCVGCGCV", "Ginny", expected);
+}
+
+static void test_full_garden_for_harriet(void)
+{
+   const plant_t expected[] = { VIOLETS, RADISHES, RADISHES, VIOLETS };
+   test_kindergarten_garden(
+       "VRCGVVRVCGGCCGVRGCVCGCGV\nVRCCCGCRRGVCGCRVVCVGCGCV", "Harriet",
+       expected);
+}
+
+static void test_full_garden_for_ileana(void)
+{
+   const plant_t expected[] = { GRASS, CLOVER, VIOLETS, CLOVER };
+   test_kindergarten_garden(
+       "VRCGVVRVCGGCCGVRGCVCGCGV\nVRCCCGCRRGVCGCRVVCVGCGCV", "Ileana",
+       expected);
+}
+
+static void test_full_garden_for_joseph(void)
+{
+   const plant_t expected[] = { VIOLETS, CLOVER, VIOLETS, GRASS };
+   test_kindergarten_garden(
+       "VRCGVVRVCGGCCGVRGCVCGCGV\nVRCCCGCRRGVCGCRVVCVGCGCV", "Joseph",
+       expected);
+}
+
+static void test_full_garden_for_kincaid_second_to_last_student_s_garden(void)
+{
+   const plant_t expected[] = { GRASS, CLOVER, CLOVER, GRASS };
+   test_kindergarten_garden(
+       "VRCGVVRVCGGCCGVRGCVCGCGV\nVRCCCGCRRGVCGCRVVCVGCGCV", "Kincaid",
+       expected);
+}
+
+static void test_full_garden_for_larry_last_student_s_garden(void)
+{
+   const plant_t expected[] = { GRASS, VIOLETS, CLOVER, VIOLETS };
+   test_kindergarten_garden(
+       "VRCGVVRVCGGCCGVRGCVCGCGV\nVRCCCGCRRGVCGCRVVCVGCGCV", "Larry", expected);
+}
+
+int main(void)
+{
+   UnityBegin("test_kindergarten-garden.c");
```

Comment:
```suggestion
   UnityBegin("test_kindergarten_garden.c");
```

Reply:
Question for the maintainers: Why do we use `UnityBegin` with an explicit filename and `UnityEnd` instead of `UNITY_BEGIN()` without the filename and `UNITY_END()`?
(see https://github.com/ThrowTheSwitch/Unity/blob/master/docs/UnityConfigurationGuide.md#main)

Reply:
This is the only project where I've used Unity and it was done like this when I became a maintainer of the track. In other words, I have no idea, and I didn't realize that there was another option.

Reply:
I fixed the name now. You want me to remove it? Another PR could change the existing exercises to make things consistent.

Reply:
@ahans another PR to change this would be great.